src/pushsource/_impl/backend/registry_source.py

```diff
@@ -98,13 +100,17 @@ def _push_item_from_registry_uri(self, uri, signing_key):
             if source_uri not in self._manifests:
                 self._manifests[source_uri] = {}
             if mtype not in self._manifests[source_uri]:
-                manifest_details = get_manifest(
-                    "%s://%s" % (schema, host),
-                    repo,
-                    src_tag,
-                    manifest_types=[mtype],
-                )
-                self._manifests[source_uri][mtype] = manifest_details
+                try:
+                    manifest_details = get_manifest(
+                        "%s://%s" % (schema, host),
+                        repo,
+                        src_tag,
+                        manifest_types=[mtype],
+                    )
+                    self._manifests[source_uri][mtype] = manifest_details
+                except requests.exceptions.HTTPError as e:
+                    if e.response.status_code == 404:
+                        continue
```

Comment:
Did you want to re-raise if it was something other than a 404?

Otherwise it seems like we can just crash at the next line trying to access `self._manifests[source_uri][mtype]`.